examples/cpp/pyperf/PyPerfBPFProgram.cc

```diff
@@ -313,6 +317,43 @@ on_event(struct pt_regs* ctx) {
   event->stack_status = STACK_STATUS_ERROR;
   event->error_code = ERROR_NONE;
 
+  struct task_struct const *const task = (struct task_struct *)bpf_get_current_task();
+
+  // Get raw native user stack
+  struct pt_regs user_regs;
+
+  // ebpf doesn't allow direct access to ctx->cs, so we need to copy it
+  int cs;
+  bpf_probe_read_kernel(&cs, sizeof(cs), &(ctx->cs));
+
+  // Are we in user mode?
+  if (cs & 3) {
```

Comment:
why not use `user_mode(regs)` like earlier? I suggested to avoid the full copy but if the verifier complains, just return it and keep it the previous way (instead of reimplementing macros haha)

Reply:
As I wrote in the comment, for some reason the verifier doesn't allow access to `ctx->cs` directly. That's why I implemented it this way in the first place.
But even though this implementation is somewhat less readable, it is more efficient as it saves copying the whole struct. It worth it IMO.
If you don't agree I'll revert it.

Reply:
Can keep it then. I preferred the readable one